app/experimenter/visualization/tests/api/test_views.py

```diff
@@ -125,14 +126,23 @@ def test_analysis_results_view_data(self, status, mock_exists, mock_open):
             "weekly": FORMATTED_DATA_WITHOUT_POPULATION_PERCENTAGE,
             "overall": FORMATTED_DATA_WITH_POPULATION_PERCENTAGE,
             "other_metrics": {"some_count": "Some Count"},
+            "metadata": {},
             "show_analysis": False,
         }
 
         class File:
+            def __init__(self, filename):
+                self.name = filename
+
             def read(self):
+                if "metadata" in self.name:
+                    return "{}"
                 return json.dumps(DATA_WITHOUT_POPULATION_PERCENTAGE)
 
-        mock_open.return_value = File()
+        def open_file(filename):
+            return File(filename)
```

Comment:
Oh you know if we're implementing this interface ourself here we can explicitly pass in a parameter to our `File` class that indicates if it's a metadata file and return `{}` isntead of checking the filepath?

Reply:
I'm not sure what you mean. I think in order to pass in a param to `File` to indicate if it's a metadata file, we would need to know if it's a metadata file. And the only way to know is by checking `if "metadata" in filename` which would just move the check from `read()` to `open_file()`.